facefusion/face_analyser.py

```diff
@@ -152,15 +167,35 @@ def detect_with_retinaface(temp_frame : Frame, temp_frame_height : int, temp_fra
 			for bbox in distance_to_bbox(anchors, bbox_raw)[keep_indices]:
 				bbox_list.append(numpy.array(
 				[
-					bbox[0] * ratio_width,
-					bbox[1] * ratio_height,
-					bbox[2] * ratio_width,
-					bbox[3] * ratio_height
+					(bbox[0] - offset_width) * ratio_width,
+					(bbox[1] - offset_height) * ratio_height,
+					(bbox[2] - offset_width) * ratio_width,
+					(bbox[3] - offset_height) * ratio_height
 				]))
 			for kps in distance_to_kps(anchors, kps_raw)[keep_indices]:
-				kps_list.append(kps * [ ratio_width, ratio_height ])
+				kps_list.append((kps - [offset_width, offset_height]) * [ ratio_width, ratio_height ])
 			for score in detections[index][keep_indices]:
 				score_list.append(score[0])
+	for index, feature_stride in enumerate(feature_strides):
+		keep_indices = numpy.where(detections_redused[index] >= facefusion.globals.face_detector_score)[0]
+		if keep_indices.any():
+			stride_height = face_detector_height // feature_stride
+			stride_width = face_detector_width // feature_stride
+			anchors = create_static_anchors(feature_stride, anchor_total, stride_height, stride_width)
+			bbox_raw = detections_redused[index + feature_map_channel] * feature_stride
+			kps_raw = detections_redused[index + feature_map_channel * 2] * feature_stride
+			for bbox in distance_to_bbox(anchors, bbox_raw)[keep_indices]:
+				bbox_list.append(numpy.array(
+				[
+					max(0, ((bbox[0] - reduction_offset_width) * reduction_ratio_width - offset_width) * ratio_width),
```

Comment:
there is too much complexity here... you calculate the dimensions from too many factors